src/mscorlib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -306,12 +308,17 @@ public override int GetHashCode(string obj)
             if (obj == null) return 0;
             return obj.GetLegacyNonRandomizedHashCode();
         }
+
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            // We are doing this to stay compatible with .NET Core 2.0 serialization payloads.
```

Comment:
This command is not correct. This part is done to be compatible with .NET Framework.